File: 3105_longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.c
int max(int a,int b){
    return a>b?a:b;
}

int longestMonotonicSubarray(int* nums, int numsSize) {
    int SI=1,SD=1,tmpI=1,tmpD=1;
    for(int i=1;i<numsSize;i++){
        if(nums[i]>nums[i-1]){
            tmpI++;
            tmpD=1;
        }else if(nums[i]<nums[i-1]){
            tmpD++;
            tmpI=1;
        }else{
            tmpI=1;
            tmpD=1;
        }
        SI=max(tmpI,SI);
        SD=max(tmpD,SD);
    }
    return max(SI,SD);
}

File: 3105_longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.java
class Solution {
    public int longestMonotonicSubarray(int[] nums) {

        int maxLength = 1;
        int incLength = 1; // 計算嚴格遞增子陣列的長度
        int decLength = 1; // 計算嚴格遞減子陣列的長度

        for (int i = 1; i < nums.length; i++) {
            if (nums[i] > nums[i - 1]) {
                incLength++; // 遞增長度 +1
                decLength = 1; // 遞減長度重置
            } else if (nums[i] < nums[i - 1]) {
                decLength++; // 遞減長度 +1
                incLength = 1; // 遞增長度重置
            } else {
                // 如果相等，兩者都重置為 1
                incLength = 1;
                decLength = 1;
            }
            maxLength = Math.max(maxLength, Math.max(incLength, decLength));
        }

        return maxLength;
    }
}